File: 034_find_first_and_last_position_of_element_in_sorted_array/Solution1/Solution.py
class Solution:
    def searchRange(self, nums: list, target: int) -> list:
        if not nums or len(nums) == 0: return [-1, -1]
        arr_len = len(nums)
        left , right = 0, arr_len - 1

        while left <= right:
            mid = (left + right) // 2
            if nums[mid] == target:
                while nums[left] != target: left += 1
                while nums[right] != target: right -= 1
                return [left, right]
            if nums[mid] < target: left = mid + 1
            else: right = mid - 1

        return [-1, -1]


if __name__ == '__main__':
    nums = [5, 7, 7, 8, 8, 10]
    target = 8
    solution = Solution()
    print(solution.searchRange(nums, target))

File: 034_find_first_and_last_position_of_element_in_sorted_array/Solution1/Solution.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        if (nums == null || nums.length == 0) return new int[]{-1, -1};
        int arrLen = nums.length;
        int left = 0, right = arrLen - 1;

        while (left <= right) {
            int mid = (left + right) / 2;
            if (nums[mid] == target) {
                while (nums[left] != target) left++;
                while (nums[right] != target) right--;
                return new int[]{left, right};
            }
            
            if (nums[mid] < target) left = mid + 1;
            else right = mid - 1;
        }

        return new int[]{-1, -1};
    }

    public static void main(String[] args) {
        int[] nums = new int[]{5, 7, 7, 8, 8, 10};
        int target = 8;
        Solution solution = new Solution();
        int[] result = solution.searchRange(nums, target);
        System.out.println(result[0] + " " + result[1]);
    }
}

